pkg/tfbridge/info.go

```diff
@@ -455,15 +455,35 @@ type DefaultInfo struct {
 	AutoNamed bool
 	// Config uses a configuration variable from this package as the default value.
 	Config string
-	// From applies a transformation from other resource properties.
+
+	// Deprecated. Use ComputeDefault.
 	From func(res *PulumiResource) (interface{}, error)
+
+	// ComputeDefault specifies how to compute a default value for the given property by consulting other properties
+	// such as the resource's URN. See [ComputeDefaultOptions] for all available information.
+	ComputeDefault func(opts ComputeDefaultOptions) (interface{}, error)
```

Comment:
I think it's generally useful to have a `context.Context` as a side-channel.
```suggestion
	ComputeDefault func(context.Context, ComputeDefaultOptions) (interface{}, error)
```

Reply:
Ok that's updated now. Better to have it locked in the signature even if it's context.TODO() for now in one place.